authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,237 @@ func getActionName(verb string) string {
 	var action string
-	if subRevReq.ResourceAttributes.Resource == "pods" && subRevReq.ResourceAttributes.Subresource == "exec" {
-		action = path.Join(subRevReq.ResourceAttributes.Resource, subRevReq.ResourceAttributes.Subresource, "action")
+
+	if resource == PodsResource && subResource == "exec" {
+		action = path.Join(resource, subResource, "action")
 	} else {
-		action = path.Join(subRevReq.ResourceAttributes.Resource, getActionName(subRevReq.ResourceAttributes.Verb))
+		action = path.Join(resource, getActionName(verb))
 	}
 
 	return action
 }
 
-func getDataAction(subRevReq *authzv1.SubjectAccessReviewSpec, clusterType string) AuthorizationActionInfo {
-	authInfo := AuthorizationActionInfo{
-		IsDataAction: true,
-	}
+func getDataActions(subRevReq *authzv1.SubjectAccessReviewSpec, clusterType string, operationsMap azureutils.OperationsMap) ([]azureutils.AuthorizationActionInfo, error) {
+	var authInfoList []azureutils.AuthorizationActionInfo
+	var err error
 
-	authInfo.AuthorizationEntity.Id = clusterType
 	if subRevReq.ResourceAttributes != nil {
-		if subRevReq.ResourceAttributes.Group != "" {
-			authInfo.AuthorizationEntity.Id = path.Join(authInfo.AuthorizationEntity.Id, subRevReq.ResourceAttributes.Group)
-		}
+		if subRevReq.ResourceAttributes.Resource != "*" && subRevReq.ResourceAttributes.Group != "*" && subRevReq.ResourceAttributes.Verb != "*" {
+			/*
+				  This sections handles the following scenarios:
+
+				    | Scenario                      | Namespace is empty (Cluster scope call)      | Namespace is not empty (NS scope)        |
+					 ------------------------------- ---------------------------------------------- ------------------------------------------
+					| Verb, Res and Group are not * | Normal single resource call at cluster scope | Normal single resource call  at ns scope |
+
+			*/
+			authInfoSingle := azureutils.AuthorizationActionInfo{
+				IsDataAction: true,
+			}
+
+			authInfoSingle.AuthorizationEntity.Id = clusterType
 
-		action := getResourceAndAction(subRevReq)
-		authInfo.AuthorizationEntity.Id = path.Join(authInfo.AuthorizationEntity.Id, action)
+			if subRevReq.ResourceAttributes.Group != "" {
+				authInfoSingle.AuthorizationEntity.Id = path.Join(authInfoSingle.AuthorizationEntity.Id, subRevReq.ResourceAttributes.Group)
+			}
+
+			action := getResourceAndAction(subRevReq.ResourceAttributes.Resource, subRevReq.ResourceAttributes.Subresource, subRevReq.ResourceAttributes.Verb)
+			authInfoSingle.AuthorizationEntity.Id = path.Join(authInfoSingle.AuthorizationEntity.Id, action)
+			authInfoList = append(authInfoList, authInfoSingle)
+
+		} else {
+			if operationsMap == nil {
+				return nil, errors.Errorf("Wildcard support for Resource/Verb/Group is not supported for request Group: %s, Resource: %s, Verb: %s", subRevReq.ResourceAttributes.Group, subRevReq.ResourceAttributes.Resource, subRevReq.ResourceAttributes.Verb)
```

Comment:
Wildcard support for Resource/Verb/Group is not enabled?

Reply:
done